### PR TITLE
[vLLM] Fix qwen3-32b batch-32 TP prefill sharding collapse; enable bs32 benchmark

### PR DESCRIPTION
**Pull request by ssaliceTT**
### Ticket
#5032

### Problem description
The qwen3-32b vLLM benchmark fails at **batch 32** on a 1×4 tensor-parallel mesh (QB2 / 4× Blackhole).

The prefill KV-cache fill loop materializes a fresh `torch.tensor([batch_idx])` per batch, per K/V, per layer. `torch` unrolls this Python loop, lifting `batch_size × 2 × num_layers` scalar `int32` constants as graph parameters, which is ~4096 for batch 32 / 64 layers. That pushes the prefill graph's parameter count over torch_xla's parameter-wrapping threshold, so it is emitted as a thin public `@main` wrapper forwarding to a private inner function. The compiler inlines that callee but does not copy the per-argument `sdy.sharding` up to `@main`, so every entry argument is treated as `Unsharded`. The compiler then inserts a replicate `distribute_tensor` on inputs that are already mesh-distributed:

- **trace off:** runtime fatal `Can't get a single buffer from host storage distributed over mesh shape MeshShape([1,4])`
- **trace on:** `TTNNLayoutAttr` `Memory layout is required for non-SystemMemory buffer type` assertion

(decode is unaffected - at batch 1 these scalars are limited as it scales based on batch_size from my tests, hence the bs1 test has not been failing on CI.)

### What's changed
- `integrations/vllm_plugin/vllm_tt/attention.py`: hoist the per-user fill batch index into a single `torch.arrange(inputs.users)` and static-slice it (`fill_batch_idx[i:i+1]`) instead of building a `torch.tensor([batch_idx])` each iteration. A static slice lowers to compile-time slice attributes, not lifted operands, so no per-user constants are added to the graph. Prefill drops back under the parameter-wrapping threshold, the graph stays flat, `sdy.sharding` survives, and both failure modes above are resolved. `paged_fill_cache` is still single-user per call; this is purely about how the index is passed.
- `tests/benchmark/test_vllm_benchmarks.py`: convert `qwen3-32b-qb2-tp` from batch 1 to batch 32 (`use_2d_mesh=False`, `gpu_memory_utilization=0.1`) so the bs32 path is exercised.

Validated on QB2 (1×4 Blackhole) locally, will add CI test.

### Checklist
- [x] qb2 w/ 1x4 mesh benchmark: https://github.com/tenstorrent/tt-xla/actions/runs/27039167668
- [ ] n300-llmbox w/ 1x8 mesh benchmark: https://github.com/tenstorrent/tt-xla/actions/runs/27039142903
- [ ] Ensure falcon 3 7B and 10B is fixed from above llmbox test.

_Note: the branch used for the tests was a custom branch with the fix but with all tests set to `use_2d_mesh=False` for illustrative purposes to show proof of support. If needed we can set them in a later PR._

### CODE REVIEW

File: integrations/vllm_plugin/vllm_tt/attention.py [hunk 1]
```diff
@@ -487,22 +487,35 @@ def _handle_paged_attention(
...
+            # total). At batch 32 / 64 layers that is ~4096 extra params, which
+            # trips torch_xla's parameter-wrapping threshold: the prefill graph
+            # is then emitted as a @main wrapper forwarding to an inner function,
+            # and the inliner drops the per-arg sdy.sharding, so already-mesh-
+            # distributed inputs get a spurious replicate distribute_tensor
+            # ("single buffer from host storage" fatal, trace off) or trip the
+            # TTNNLayout assertion (trace on). See issue #5032. The static slice
+            # fill_batch_idx[i:i+1] lowers to compile-time slice attributes, not
+            # lifted operands, so the prefill graph stays flat.
+            fill_batch_idx = torch.arange(
```
Comment (AleksKnezevic) [inline integrations/vllm_plugin/vllm_tt/attention.py:504]: Are we able to create this input once at setup or does it need to be done on each iteration? Also, I think you're better off creating it on 'cpu' and moving it to 'device'

File: integrations/vllm_plugin/vllm_tt/attention.py [hunk 2]
```diff
@@ -487,22 +487,35 @@ def _handle_paged_attention(
             key_for_update = inputs.key.transpose(1, 2)
             value_for_update = inputs.value.transpose(1, 2)
 
+            # Hoist the per-user batch index into a single tensor and
```
Comment (AleksKnezevic) [inline integrations/vllm_plugin/vllm_tt/attention.py:490]: I don't know how others feel about this, so would love your opinion @mmanzoorTT and @kmabeeTT, but I think long comments like this belong in the PR description, not necessarily the code base.

Comment (ssaliceTT) [reply integrations/vllm_plugin/vllm_tt/attention.py:490]: I don't disagree, the vllm_tt files tend to be comment-heavy, so I matched the existing pattern, but you're right that the mechanism write-up belongs in the PR description (and it's already there). I'll trim in-code comments moving on.

**Comment by ssaliceTT** [comment]
@AleksKnezevic @mmanzoorTT putting this PR on hold for now.
It was meant to be a quick fix ahead of a better solution with downstream tt-mlir changes, as described in my comment on #5032 and verified locally with a quick POC of a batched paged_fill_cache (my tt-mlir branch `ssalice/batched-paged-fill-cache`).
From my convo with @kmabeeTT today we've figured out that this should come with #4955 once the tt-mlir uplift goes through, and I believe it should fix this bug. Will revisit if needed.

File: integrations/vllm_plugin/vllm_tt/attention.py (hunk 2 above)
Comment (kmabeeTT) [reply integrations/vllm_plugin/vllm_tt/attention.py:490]: Agreed in this case yeah, though in general I am fine with a 2 line comment if it documenting something non obvious about behavior since very few people will be browsing old PR's while looking at code. I usually tell claude to do a pass to make all remaining comments minimal and concise if any truly must remain.

File: integrations/vllm_plugin/vllm_tt/attention.py (hunk 1 above)
Comment (mmanzoorTT) [reply integrations/vllm_plugin/vllm_tt/attention.py:504]: We are always using same number of users for generative models; so we can create this input once; applying batch padding for fewer requests. I have another branch where I am passing `batch_size` as `attn_metadata` attribute; we can use `batch_size` to create `fill_batch_idx` during initialization instead of runtime. I can cleanup the code and @ssaliceTT can cherry-pick and try it out.
